src/libraries/Microsoft.PowerFx.Interpreter/Functions/IRandomService.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerFx.Functions
+{
+    /// <summary>
+    /// Service for any functions needing Random behavior, 
+    /// such as Rand(), RandBetween(), or Shuffle().
+    /// </summary>
+    public interface IRandomService
+    {
+        /// <summary>
+        /// Get the next random number. 
+        /// </summary>
+        /// <returns></returns>
+        double NextDouble();
+    }
+
+    // Default implementation of IRandomService
+    internal class DefaultRandomService : IRandomService
```

Comment:
>DefaultRandomService [](http://example.com/codeflow?start=19&length=20)

I would add a comment
Not suitable for any security related purpose.

See [https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca5394](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca5394)